riscv: tlx: fix retention .

haiwentelink opened the pull request:
- fix link to put retention_reset to the first .
- change start.s for ble ramcode section.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/8ebb1d921f32051114c2ae3a7c0d651a1e983bd9 | https://github.com/telink-semi/hal_telink/commit/63c96047ba8f0cf56976d8d6d63e7641a5eea05a (develop_customer) | [telink-semi/hal_telink@8ebb1d92..63c96047](https://github.com/telink-semi/hal_telink/compare/8ebb1d921f32051114c2ae3a7c0d651a1e983bd9..63c96047ba8f0cf56976d8d6d63e7641a5eea05a) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

haiwentelink committed:
riscv: tlx: fix retention .

- fix link to put retention_reset to the first .
- change start.s for ble ramcode section.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
